receiver/apachereceiver/config.go

```diff
@@ -55,5 +59,16 @@ func (cfg *Config) Validate() error {
 	}
 
 	cfg.serverName = u.Hostname()
+	cfg.port = u.Port()
+
+	if cfg.port == "" {
+		if u.Scheme == "https" {
+			cfg.port = httpsDefaultPort
+		} else if u.Scheme == "http" {
+			cfg.port = httpDefaultPort
+		}
```

Comment:
The `Validate` function should not change the configuration.

I suggest moving the `port` field to the scraper and setting it in the factory.

Reply:
I understand the reason, but `serverName` is set in this function anyway - should it also be changed?

Reply:
Yeah, really it should. I think it's an inconsequential change but is better for the sake of consistency, so it's fine to do in this PR in my opinion.

Reply:
I think we have caught a bug here.
After the fix, the integration tests stopped passing - the expected value of `serverName` was `""`, while the receiver returned `"localhost"`. I checked immediately - and it happens that `Validate` is called only inside unit tests - so outside the tests, `serverName` wasn't actually set and was always an empty string. 
I fixed this and pushed the changes on this branch.